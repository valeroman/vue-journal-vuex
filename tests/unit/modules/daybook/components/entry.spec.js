import Entry from '@/modules/daybook/components/Entry'
import { shallowMount } from '@vue/test-utils'
import { journalState } from '../../../mock-data/test-journal-state'

describe('Pruebas en el Entry Component', () => {

    // mockRouter
    const mockRouter = {
        push: jest.fn()
    }

    const wrapper = shallowMount(Entry, {
        props: {
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    })


    test('debe de hacer match con el snapshot', () => {

        expect(wrapper.html()).toMatchSnapshot()
    })

    test('debe de redireccionar al hacer click en el entry-container', () => {

        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')

        expect(mockRouter.push).toHaveBeenCalledWith({
            name: 'entry',
            params: {
                id: journalState.entries[0].id
            }
        })
    })

    test('pruebas en la propiedades computadas', () => {

        // wrapper.vm <========== ver las propiedades computadas
        // console.log(wrapper.vm.day)
        // console.log(wrapper.vm.month)
        // console.log(wrapper.vm.yearDay)

        expect(wrapper.vm.day).toBe(9)
        expect(wrapper.vm.month).toBe('Diciembre')
        expect(wrapper.vm.yearDay).toBe('2021, Jueves')
    })



})