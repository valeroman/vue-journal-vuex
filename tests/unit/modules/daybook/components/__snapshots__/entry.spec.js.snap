// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry Component debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- titulo -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">9</span><span class="mx-1 fs-5">Diciembre</span><span class="mx-2 fw-light">2021, Jueves</span></div><!-- descripcion -->
  <div class="entry-description">Hola desde el mock data</div>
</div>
`;
